apps/theming-designer/src/components/SemanticSlots.tsx

```diff
@@ -432,33 +58,107 @@ export const SemanticSlots: React.StatelessComponent<ISemanticSlotsProps> = (pro
+        palette === 'white' ||
+        palette === 'redDark' ||
+        palette === 'accent'
+      ) {
+        trimmedPaletteSlots[palette] = (paletteSlots as any)[palette];
+      }
+    }
+    return trimmedPaletteSlots;
+  };
+
+  const isASemanticColor = (semanticString: string): boolean => {
+    return (
+      semanticString.startsWith('body') ||
+      semanticString.startsWith('primary') ||
+      semanticString.startsWith('disabled') ||
+      semanticString.startsWith('focus') ||
+      semanticString.startsWith('variant') ||
+      semanticString.startsWith('default') ||
+      semanticString.includes('action') ||
+      semanticString.includes('link') ||
+      semanticString.startsWith('accent') ||
+      semanticString.startsWith('list') ||
+      semanticString.startsWith('button') ||
+      semanticString.startsWith('menu') ||
+      semanticString.startsWith('input') ||
+      semanticString.startsWith('small')
+    );
+  };
+
+  const trimSemanticSlotsOrNames = <T extends unknown>(semanticSlots: T) => {
+    if (semanticSlots instanceof Array) {
+      let trimmedSemanticSlotNames = [];
+      for (let i = 0; i < semanticSlots.length; i++) {
+        let slotName = semanticSlots[i];
+        if (isASemanticColor(slotName)) {
+          trimmedSemanticSlotNames.push(slotName);
+        }
+      }
+      return trimmedSemanticSlotNames;
+    } else {
+      let trimmedSemanticSlots: ISemanticSlots = {};
+      for (let semanticColor in semanticSlots) {
+        if (isASemanticColor(semanticColor)) {
+          trimmedSemanticSlots[semanticColor] = (semanticSlots as any)[semanticColor];
+        }
+      }
+      return trimmedSemanticSlots;
+    }
+  };
+
+  const fillVariantSlotsList = (variantType: VariantThemeType): JSX.Element[] => {
     let currThemeVariant: ITheme;
+    let noneVariant = getVariant(props.theme, VariantThemeType.None);
     if (props.theme) {
       currThemeVariant = getVariant(props.theme, variantType);
-      let currVariantSemanticColors = currThemeVariant.semanticColors;
-      let tempJSXList: JSX.Element[];
-      tempJSXList = [];
+      const currVariantSemanticSlots = currThemeVariant.semanticColors;
+      // "trimming" to get rid of the seamantic color slots & palette slots we don't use for theme designer app
+      const trimmedSemanticSlots = trimSemanticSlotsOrNames(currVariantSemanticSlots);
+      const currVariantPaletteSlots = noneVariant.palette; // palette slot values should be based off the default variant
+      const trimmedPaletteSlots = trimPaletteSlots(currVariantPaletteSlots);
+      const mapping: IMapping = {};
+      // Iterate through the list of semantic colors
+      // for each semantic color, check if it's hex color string is in the list of palette colors
+      // if it is, add it to the mapping
+      for (let semanticColor in trimmedSemanticSlots) {
+        if (semanticColor) {
+          const paletteColorHexStr = (trimmedSemanticSlots as any)[semanticColor];
```

Comment:
you shouldn't need this any

Reply:
When I remove it, it says "type 'any[] | ISemanticSlots' has no index signature" so I think I do need it... I've been trying to find ways to remove all usage of any and it's been a struggle.

Reply:
Nvm done - removed all any usage!